pkg/apis/autoscaling/types.go

```diff
@@ -420,6 +471,20 @@ type MetricStatus struct {
 	// to normal per-pod metrics using the "pods" source.
 	// +optional
 	Resource *ResourceMetricStatus
+	// PodResource refers to a resource metric (such as those specified in
+	// requests and limits) known to Kubernetes describing each pod in the
+	// current scale target (e.g. CPU or memory). Such metrics are built in to
+	// Kubernetes, and have special scaling options on top of those available
+	// to normal per-pod metrics using the "pods" source.
+	// +optional
+	PodResource *PodResourceMetricStatus
+	// ContainerResource refers to a resource metric (such as those specified in
+	// requests and limits) known to Kubernetes describing a single container in each pod in the
+	// current scale target (e.g. CPU or memory). Such metrics are built in to
+	// Kubernetes, and have special scaling options on top of those available
+	// to normal per-pod metrics using the "pods" source.
+	// +optional
+	ContainerResource *ContainerResourceMetricStatus
```

Comment:
the same feature-gate would protect this field

Reply:
in autoscalerStatusStrategy#PrepareForUpdate, follow the pattern of clearing the field if the feature is off AND the existing object does not use the new field

Reply:
Updating the status is not allowed in the strategy. So for cases when the feature gate if off it will simply get copied over from the old status if it exists. And it can only exist if the controller puts it there because the feature was enabled and used.